app/experimenter/kinto/tasks/nimbus.py

```diff
@@ -0,0 +1,164 @@
+import markus
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from experimenter.celery import app
+from experimenter.experiments.api.v6.serializers import NimbusExperimentSerializer
+from experimenter.experiments.changelog_utils import generate_nimbus_changelog
+from experimenter.experiments.models import (
+    NimbusBucketRange,
+    NimbusExperiment,
+    NimbusIsolationGroup,
+)
+from experimenter.kinto.client import KintoClient
+
+logger = get_task_logger(__name__)
+metrics = markus.get_metrics("kinto.tasks")
+
+
+def get_kinto_user():
+    user, _ = get_user_model().objects.get_or_create(
+        email=settings.KINTO_DEFAULT_CHANGELOG_USER,
+        username=settings.KINTO_DEFAULT_CHANGELOG_USER,
+    )
+    return user
+
+
+@app.task
+@metrics.timer_decorator("push_experiment_to_kinto.timing")
+def nimbus_push_experiment_to_kinto(experiment_id):
+    kinto_client = KintoClient(settings.KINTO_COLLECTION_NIMBUS)
+
+    metrics.incr("push_experiment_to_kinto.started")
+
+    experiment = NimbusExperiment.objects.get(id=experiment_id)
+    if not NimbusBucketRange.objects.filter(experiment=experiment).exists():
+        NimbusIsolationGroup.request_isolation_group_buckets(
+            experiment.slug,
+            experiment,
+            NimbusExperiment.BUCKET_COUNT,
+        )
+
+    data = NimbusExperimentSerializer(experiment).data
+
+    logger.info(f"Pushing {experiment} to Kinto")
+
+    try:
+        kinto_client.push_to_kinto(data)
+
+        experiment.status = NimbusExperiment.Status.ACCEPTED
+        experiment.save()
+
+        generate_nimbus_changelog(experiment, get_kinto_user())
+
+        logger.info(f"{experiment} pushed to Kinto")
+        metrics.incr("push_experiment_to_kinto.completed")
+    except Exception as e:
+        metrics.incr("push_experiment_to_kinto.failed")
+        logger.info(f"Pushing {experiment} to Kinto failed: {e}")
+        raise e
+
+
+@app.task
+@metrics.timer_decorator("check_kinto_push_queue")
+def nimbus_check_kinto_push_queue():
+    kinto_client = KintoClient(settings.KINTO_COLLECTION_NIMBUS)
+
+    metrics.incr("check_kinto_push_queue.started")
+
+    queued_experiments = NimbusExperiment.objects.filter(
+        status=NimbusExperiment.Status.REVIEW
+    )
```

Comment:
This portion of the code makes it appear that this function is going to be working with multiple queued experiments. However the rejected code and the portion below where it actually pushes an experiment, only works with a single experiment at a time. This function would greatly benefit from a detailed doc string that indicates the desired method of operation of this task and reasoning behind it.